core/commands/object/object.go

```diff
@@ -381,6 +383,16 @@ And then run:
 			return
 		}
 
+		dopin, _, err := req.Option("pin").Bool()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		if dopin {
+			defer n.Blockstore.PinLock().Unlock()
+		}
+
 		output, err := objectPut(n, input, inputenc, datafieldenc)
```

Comment:
could we make `objectPut` just return a cid? It returns an `Object` type for some reason that contains a stringed cid and an array of links for some reason. It doesnt ever use the links, seems like we could make this a little cleaner